File: routes/objc-to-javascript/corpus/module/equivalence_module.m
#import <Foundation/Foundation.h>

long long calculate(long long subtotal, long long tax) {
    if (subtotal < 0LL) {
        return 0LL;
    }
    return subtotal + tax;
}

long long clamp(long long value, long long minimum, long long maximum) {
    if (value < minimum) {
        return minimum;
    }
    if (value > maximum) {
        return maximum;
    }
    return value;
}

long long difference(long long left, long long right) {
    return left - right;
}

double clampNumber(double value, double minimum, double maximum) {
    if (value < minimum) {
        return minimum;
    }
    if (value > maximum) {
        return maximum;
    }
    return value;
}

BOOL both(BOOL left, BOOL right) {
    return left && right;
}

File: routes/php-to-javascript/corpus/module/equivalence_module.php
<?php

declare(strict_types=1);

function calculate(int $subtotal, int $tax): int {
    if ($subtotal < 0) { return 0; }
    return $subtotal + $tax;
}

function clamp(int $value, int $minimum, int $maximum): int {
    if ($value < $minimum) { return $minimum; }
    if ($value > $maximum) { return $maximum; }
    return $value;
}

function difference(int $left, int $right): int { return $left - $right; }

function clampNumber(float $value, float $minimum, float $maximum): float {
    if ($value < $minimum) { return $minimum; }
    if ($value > $maximum) { return $maximum; }
    return $value;
}

function both(bool $left, bool $right): bool { return $left && $right; }
